test_util/summaryComparator.cpp

```diff
@@ -0,0 +1,203 @@
+    std::cout << "Suggested combination of command line options"<< std::endl; 
+    std::cout << " -i -g -m mainVariable, will print the vector which have the greatest error ratio of the mainVariable of interest "<< std::endl; 
+    
+
+}
+
+//---------------------------------------------------
+
+
+int main (int argc, char ** argv){
+    
+    //------------------------------------------------
+    //Defines some constants
+    bool specificKeyword = false;
+    bool allowSpikes = false;
+    bool findVolumeError = false;
+    bool integrationTest = false;
+    bool regressionTest = true; 
+    bool allowDifferentAmountOfKeywords = true;
+    bool printKeywords = false;
+    bool printSpecificKeyword = false; 
+    bool findVectorWithGreatestErrorRatio = false; 
+    bool oneOfTheMainVariables = false; 
+    bool throwExceptionForTooGreatErrorRatio = true;
+    const char* keyword  = nullptr;
+    const char* mainVariable = nullptr; 
+    int c = 0;
+    int limit = -1;
+    //------------------------------------------------
+
+    //------------------------------------------------
+    //For setting the options selected 
+    while ((c = getopt(argc, argv, "dghik:m:KpP:rs:vV:")) != -1) {
+        switch (c) {
+	case 'd':
+	    throwExceptionForTooGreatErrorRatio = false;
+	    break;
+	case 'g':
+	    findVectorWithGreatestErrorRatio = true;
+	    throwExceptionForTooGreatErrorRatio = false;
+	    break;
+	case 'h':
+	    printHelp();
+	    return 0;
+	case 'i':
+	    integrationTest = true; 
+	    regressionTest = false;
+	    break;
+	case 'k':
+	    specificKeyword = true;
+	    keyword = optarg;
+	    break;
+	case 'K':
+	    allowDifferentAmountOfKeywords = false; 
+	    break; 
+	case 'm':
+	    oneOfTheMainVariables = true;
+	    mainVariable = optarg;
+	    break;
+	case 'p':
+	    printKeywords = true; 
+	    break; 
+	case 'P':
+	    specificKeyword = true;
+	    printSpecificKeyword = true; 
+	    keyword = optarg;
+	    break;
+	case 'r':
+	    integrationTest = false; 
+	    regressionTest = true;
+	    break;
+	case 's':
+	    allowSpikes = true;
+	    limit = atof(optarg);      
+	    break;
+	case 'v':
+	    findVolumeError = true; 
+	    break; 
+	case 'V':
+	    findVolumeError = true; 
+	    specificKeyword = true;
+	    keyword = optarg;
+	    break;
+	case '?':
+	    if (optopt == 'k' || optopt == 'm' || optopt == 's') {
+		std::cout << "Option requires an keyword." << std::endl;
+		return EXIT_FAILURE;
+	    }
+	    else {
+		std::cout << "Unknown option." << std::endl;
+		return EXIT_FAILURE;
+	    }
+	default:
+	    return EXIT_FAILURE;
+        }
+    }
+    //------------------------------------------------
+    
+    
+    int argOffset = optind;
+    if (argc != argOffset + 4) {
+        printHelp();
+        return EXIT_FAILURE;
+    }
+    const char * basename1   = argv[argOffset];
+    const char * basename2   = argv[argOffset+1];
+    double absoluteTolerance = atof(argv[argOffset+2]);
```

Comment:
`atof` is quite fragile when it comes to malformed input - maybe consider e.g. `strtod`